app/src/main-process/menu/build-default-menu.ts

```diff
@@ -16,12 +16,21 @@ const defaultShellLabel = __DARWIN__
 const defaultPullRequestLabel = __DARWIN__
   ? 'Create Pull Request'
   : 'Create &pull request'
+const defaultBranchNameValue = __DARWIN__ ? 'Default Branch' : 'default branch'
 
-export function buildDefaultMenu(
-  editorLabel: string = defaultEditorLabel,
-  shellLabel: string = defaultShellLabel,
-  pullRequestLabel: string = defaultPullRequestLabel
-): Electron.Menu {
+export type MenuLabels = {
+  editorLabel?: string
+  shellLabel?: string
+  pullRequestLabel?: string
+  defaultBranchName?: string
+}
+
+export function buildDefaultMenu({
+  editorLabel = defaultEditorLabel,
+  shellLabel = defaultShellLabel,
+  pullRequestLabel = defaultPullRequestLabel,
+  defaultBranchName = defaultBranchNameValue,
+}: MenuLabels = {}): Electron.Menu {
```

Comment:
I think we can do away with the `= {}` usage in here and ensure `main.ts` line 203 passes in an empty object:

```ts
  let menu = buildDefaultMenu({})
```

Reply:
I there any reasoning behind this recommendation besides improved readability? Readability is a subjective thing and sort of hard to follow.

Reply:
We generally try to avoid optional parameters with defaults in TS, because it can be used to obscure what the API is capable of. There's already a fair bit of complexity here around destructuring this parameter and providing default values for it, so I settled on requiring the caller to provide an empty object.

If there were lots of callers that needed to do this I'd likely spend more time figuring out the right API to support them, but as it's one caller (and called exactly once during startup) I was fine with dropping the optional part.

Reply:
@a-golovanov also, here's some extra context about why we like to avoid these implicit parameters: https://github.com/desktop/desktop/pull/5979#discussion_r227826267